activity/nextflow/main.nf

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env nextflow
+
+params.input = "${projectDir}/data/*.png"
+params.prompts = 'a photo of a cat,a photo of a dog,a photo'
+params.outdir = 'results'
+
+workflow {
+    pics = Channel.fromPath(params.input)
+
+    Classify(pics, params.prompts)
+    | view
+    // | map { label, pic -> [ label.text.trim(), pic ] }
+    // | Resize
+    // | groupTuple
+    // | Collage
+    // | collect
+    // | CombineImages
+}
+
+process Classify {
+    memory '8G'
+    container 'robsyme/clip-cpu:1.0.0'
+
+    input:
+    path pic
+    val prompts
+
+    output:
+    tuple path("out.txt"), path(pic)
```

Comment:
The only cost to getting rid of the `map` operator is that when the participants view the contents of the channel, it will have the newline included:

```
[a photo of a dog
, chihuahua.png]
[a photo of a dog
, dog.png]
```

The workflow handles it nicely though because Nextflow trims the lines for us when interpolating the script block.

Reply:
This does look very neat though, so I think it's probably worth it:

```groovy
workflow {
    pics = Channel.fromPath(params.input)

    Classify(pics, params.prompts)
    | Resize
    | groupTuple
    | Collage
    | collect
    | CombineImages
}
```